File: src/pocketcloud/event/impl/server/ServerEvent.php
<?php

namespace pocketcloud\event\impl\server;

use pocketcloud\event\Event;
use pocketcloud\server\CloudServer;

abstract class ServerEvent extends Event {

    public function __construct(private CloudServer $server) {}

    public function getServer(): CloudServer {
        return $this->server;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/event/impl/server/ServerEvent.java
package de.pocketcloud.cloud.event.impl.server;

import de.pocketcloud.cloud.event.CloudEvent;
import de.pocketcloud.cloud.server.CloudServer;
import lombok.Getter;

public abstract class ServerEvent extends CloudEvent {

    @Getter
    private final CloudServer server;

    public ServerEvent(CloudServer server) {
        this.server = server;
    }
}